src/qibolab/pulses/pulse.py

```diff
@@ -111,23 +111,20 @@ def phase(self) -> float:
     def id(self) -> int:
         return id(self)
 
-    def envelope_waveform_i(self, sampling_rate=SAMPLING_RATE) -> Waveform:
+    def i(self, times: Times) -> Waveform:
         """The envelope waveform of the i component of the pulse."""
 
-        return self.shape.envelope_waveform_i(sampling_rate)
+        return self.amplitude * self.envelope.i(times)
```

Comment:
Unfortunately, my knowledge on the field is limited and to be honest I do not fully understand the references I find online but I suspect that the way we are generating `i` and `q` is not correct because we are ignoring the `pulse.relative_phase`. What I have in mind is something like
```suggestion
        return self.amplitude * (self.envelope.i(times) * np.cos(self.relative_phase) - self.envelope.q(times) * np.sin(self.relative_phase))
```
but I would feel much more confident about this if we check with some reference (which we should do if we decide to implement).

This is obviously not caused by this PR and if I am right the issue also exists in 0.1, but since we are doing related updates here, I thought it is a good place to discuss it.

The motivation behind this is that some time ago I was testing some calibration scripts from a QM repo and particularly looking at the lines here: https://github.com/qua-platform/qua-libs/blob/b7bfc54d37cb4472ac28d228db7e6e35c11fc11b/Quantum-Control-Applications/Superconducting/Single-Fixed-Transmon/configuration.py#L111
They are defining the Y180 (RY) gate using DRAG as `(I=-der, Q=gauss)`. Now this gate is basically our RX with `relative_phase=0`. So based on the equation I wrote above we would have
* RX: `relative_phase=0` -> `(I=gauss, Q=der)`
* RY: `relative_phase=pi/2` -> `(I=-der, Q=gauss)`

which seems right, but of course doesn't mean that the equation is correct for other phase values.

Reply:
I believe that part of the problem is the inconsistent use of shapes in v0.1, since many vendors' APIs are implementing their own.

My personal take is that we should move to use the vendors' APIs for custom whenever it is not rectangular, delegating the envelopes' calculation to Qibolab, to standardize more across the different drivers.

Then, Qibolab was already taking care of relative phases, but only during software modulation:
https://github.com/qiboteam/qibolab/blob/608bb8d42d0069602774d7299fa44ce22364075a/src/qibolab/pulses/modulation.py#L34

I believe this is a heritage of the Qblox drivers, and times when they were only software modulated.

In any case, the relative phase is truly a modulation feature, so I believe we should check in the drivers (since the hardware modulation is of course handled by them) how to propagate it, before implementing it on Qibolab side (just because it could be more efficient, if applied on hardware).

Reply:
@stavros11: I'm closing this conversation, as I'm not going to take action here.

> This is obviously not caused by this PR and if I am right the issue also exists in 0.1, but since we are doing related updates here, I thought it is a good place to discuss it.

As you said, it's not caused by this PR, and it is already big enough. I also believe the problem to be decoupled.

In any case, I mentioned the conversation in the OP, and there is an open item about it, possibly to be turned into an issue.

(baseline: not in this PR, so let's keep the conversations focused on related topics)